### ISSUE TITLE
Lazy src/dest closures don't seem to work

### ISSUE THREAD

**Issue by jfbibeau** [michel-kraemer/gradle-download-task#125]
Hi,

I'm having trouble passing closures as src/dest to a `Download` task, they seem to be evaluated right away. Here's a small reproducible code snippet:

```
def baseUrl = ""
def baseDest = ""

task setVariables {
    doLast {
        baseUrl = 'http://www.example.com/index.html'
        baseDest = buildDir
    }
}
task downloadFile(type: Download, dependsOn: 'setVariables') {
    src { "${baseUrl}" }
    dest { "${baseDest}" }
}
```

If `src` and `dest` are truly evaluated during execution phase, then this should work. `setVariables` should run first, set the variables, then `downloadFile` should run and have a properly set `src` and `dest`. However this fails complaining that `src` is blank (`java.net.MalformedURLException: no protocol:`).

Am I misunderstanding how passing a closure to src and dest is supposed to work? This is with Gradle 4.10.

Thanks!

**Comment by michel-kraemer** [comment]
It might be that the closures are evaluated too early. I will check that and get back to you.

**Comment by michel-kraemer** [comment]
This issue will be fixed in the upcoming version 4.0.0 of gradle-download-task.
